### PR TITLE
allow sidewalk-separate answer

### LINKED ISSUE #1925: Allow answering "sidewalk mapped separately" (sidewalk=separate) in the sidewalk quest
Via can't say menu

**Use case**
https://www.openstreetmap.org/note/2244771

**Proposed Solution**
See title (I will make pr for this after the signal that it would be accepted)

### PR DESCRIPTION
**Pull request by matkoniecz**
fixes #1925

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/quests/sidewalk/SidewalkAnswer.kt [hunk 1]
```diff
@@ -1,3 +1,5 @@
 package de.westnordost.streetcomplete.quests.sidewalk
 
-data class SidewalkAnswer(val left:Boolean, val right:Boolean)
+abstract class SidewalkAnswer
+open class SidewalkSides(val left:Boolean, val right:Boolean) : SidewalkAnswer()
+object SeparatelyMapped: SidewalkAnswer()
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/sidewalk/SidewalkAnswer.kt:5]: Please use sealed classes. Then you can delete the 
```
            else -> throw IllegalArgumentException("it should never happen")
```
in that other file

File: app/src/main/java/de/westnordost/streetcomplete/quests/sidewalk/SidewalkAnswer.kt [hunk 2]
```diff
@@ -1,3 +1,5 @@
 package de.westnordost.streetcomplete.quests.sidewalk
 
-data class SidewalkAnswer(val left:Boolean, val right:Boolean)
+sealed class SidewalkAnswer
+open class SidewalkSides(val left:Boolean, val right:Boolean) : SidewalkAnswer()
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/sidewalk/SidewalkAnswer.kt:4]:
```suggestion
data class SidewalkSides(val left:Boolean, val right:Boolean) : SidewalkAnswer()
```

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -849,5 +849,6 @@ Otherwise, you can download another keyboard in the app store. Popular keyboards
     <string name="user_profile_local_rank">Rank in\n%s</string>
     <string name="user_profile_days_active">Days\nactive</string>
     <string name="user_profile_achievement_levels">Achievement\nlevels</string>
+    <string name="quest_sidewalk_separately_mapped">Displayed separately on the map</string>
```
Comment (westnordost) [inline app/src/main/res/values/strings.xml:852]:
```suggestion
    <string name="quest_sidewalk_separately_mapped">Yes, displayed separately on map</string>
```

"Displayed separetely [...]" doesn't sound like an answer to "Does this street have a sidewalk?"

**Comment by matkoniecz** [comment]
I retested it and new text still fits.

**Pull request merged**